src/Gallery/GalleryProvider.php

```diff
@@ -0,0 +1,199 @@
+            }
+
+            return SerializerFactory::getInstance()->deserialize(
+                $response->getBody()->getContents(),
+                DynamicBackgroundOutputDto::class,
+                'json'
+            );
+        } catch (\Exception $e) {
+            throw new SherlException(SherlException::FETCH_FAILED, $e->getMessage());
+        }
+    }
+
+    public function deleteGallery(string $galleryId): ?GalleryOutputDto
+    {
+        try {
+            $response = $this->client->delete("/api/galleries/$galleryId", [
+              "headers" => [
+                "Content-Type" => "application/json",
+              ],
+            ]);
+
+            if ($response->getStatusCode() >= 300) {
+                return $this->throwSherlGalleryException($response);
+            }
+
+            return SerializerFactory::getInstance()->deserialize(
+                $response->getBody()->getContents(),
+                GalleryOutputDto::class,
+                'json'
+            );
+        } catch (\Exception $e) {
+            throw new SherlException(SherlException::FETCH_FAILED, $e->getMessage());
+        }
+    }
+
+    public function getDynamicBackgrounds(GetDynamicBackgroundFilters $filters): ?DynamicBackgroundOutputDto
+    {
+        try {
+            $response = $this->client->get('/api/galleries/dynamic-background', [
+              "headers" => [
+                "Content-Type" => "application/json",
+              ],
+              RequestOptions::QUERY => $filters,
+            ]);
+
+            if ($response->getStatusCode() >= 300) {
+                return $this->throwSherlGalleryException($response);
+            }
+
+            return SerializerFactory::getInstance()->deserialize(
+                $response->getBody()->getContents(),
+                DynamicBackgroundOutputDto::class,
+                'json'
+            );
+        } catch (\Exception $e) {
+            throw new SherlException(SherlException::FETCH_FAILED, $e->getMessage());
+        }
+
+    }
+
+    public function getGalleries(GetGalleriesFiltersDto $filters): ?GalleyOutputDto
+    {
+        try {
+            $response = $this->client->get('/api/galleries', [
+              "headers" => [
+                "Content-Type" => "application/json",
+              ],
+              RequestOptions::QUERY => $filters,
+            ]);
+
+            if ($response->getStatusCode() >= 300) {
+                return $this->throwSherlGalleryException($response);
+            }
+
+            return SerializerFactory::getInstance()->deserialize(
+                $response->getBody()->getContents(),
+                GalleyOutputDto::class,
+                'json'
+            );
+        } catch (\Exception $e) {
+            throw new SherlException(SherlException::FETCH_FAILED, $e->getMessage());
+        }
+    }
+
+    public function registerDynamicBackground(CreateDynamicBackgroundInputDto $dynamicBackground): ?DynamicBackgroundOutputDto
```

Comment:
```suggestion
    public function createDynamicBackground(CreateDynamicBackgroundInputDto $dynamicBackground): ?DynamicBackgroundOutputDto
```

"create" me parait plus adapté que "register" dans ce cas.